src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs

```diff
@@ -82,6 +82,10 @@ internal override IInvisibleEditor OpenInvisibleEditor(DocumentId documentId)
                 }
             }
 
+            // Documents in the VisualStudioWorkspace always have file paths since that's how we get files given
+            // to us from the project system.
+            Contract.ThrowIfNull(textDocument.FilePath);
```

Comment:
❔ Where would the previous implementation have thrown if this invariant was violated?

Reply:
Calling the VS API with an empty file path wouldn't have worked, that much is certain.